assets/javascripts/test_result.js

```diff
@@ -96,6 +96,27 @@ function checkPreviewVisible(stepPreviewContainer, preview) {
   }
 }
 
+function XYRes() {
+  // the test VM's resolution
+  let x_res;
+  let y_res;
+  try {
+    var xhr = new XMLHttpRequest();
+    var test_file =  "/api/v1/jobs/"+testStatus.jobid;
+    xhr.open("GET", test_file, false);
```

Comment:
The issue I had with the calls need a non-empty return, and I don't know how to do that via the async calls. Is the latter a thing?

Reply:
Let's talk about improving the AJAX code once we have established that there's no way around it. I still think it can be avoided completely, see my other comment about it.

Reply:
I think I've managed to implement your original suggestion. Please let me know what you think :)